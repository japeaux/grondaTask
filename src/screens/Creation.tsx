import React, {useEffect, useState} from "react";
import { View, Text, SafeAreaView, TouchableOpacity, Image, StatusBar, StyleSheet } from 'react-native';

import {propsStack} from '../routes/Router'
import { useNavigation, useRoute } from '@react-navigation/native'
import AsyncStorage from '@react-native-async-storage/async-storage';

export function Creation(){

    const [visits, setVisits] = useState('');

    //Reads locally stored values by the id and adds +1 to visitis every time tha page is opened
    const readData = async () => {
      try {
        const value = await AsyncStorage.getItem(params.params?.id.toString());
        if (value !== null) {
          setVisits((Number(value)+1).toString());
          saveData((Number(value)+1).toString());
        }else{
          setVisits('1');
          saveData('1');
        }
      } catch (e) {
        console.log('Failed to fetch the input from storage');
      }
    };
    
    const saveData = async (value) => {
      try {
        await AsyncStorage.setItem(params.params?.id.toString(), value)
        console.log('Data successfully saved')
      } catch (e) {
        console.log('Failed to save the data to the storage')
      }
    }

    useEffect(() => {
      readData();
    }, []);

    const navigation = useNavigation<propsStack>()
    const params = useRoute();

    return (
      
      <SafeAreaView style={style.contanier}>
       <StatusBar backgroundColor="transparent" translucent={true} barStyle='light-content'/>
        
          <TouchableOpacity onPress={()=>navigation.navigate("Home")} style={{marginTop:30}}>
            <Image
                source={require('../../assets/back.png')}
                style={style.image}  
            />
        </TouchableOpacity>

        <View style={style.subcontainer}>
          <Text style={style.text}>{visits}</Text>
          <Text style={style.text}>Visits</Text>
        </View>
        
      </SafeAreaView>

      );
}

const style = StyleSheet.create({
  contanier:{
    backgroundColor: "#28586C", 
    flex: 1
  },
  subcontainer:{
    flex:1,
    justifyContent: "center",
    alignItems: "center"
  },
  text:{
    fontSize:20, 
    fontWeight:'bold',
    color:'white'
  },
  image:{
    width: 60,
    height: 60,
    resizeMode: "contain",
  }
});
