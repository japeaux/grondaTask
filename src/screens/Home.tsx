import React from "react";
import { View, Text, SafeAreaView, ScrollView, StyleSheet, StatusBar } from 'react-native';
import Banner from "../containers/home/Banner";
import Header from "../containers/home/Header";
import CreationList from '../containers/home/CreationList'


export function Home(){
  

  return (
    <SafeAreaView style={style.contanier}>
      <StatusBar backgroundColor="transparent" translucent={true} barStyle='dark-content'/>
      <Header/>
      <ScrollView>
        <Banner/>
        <Text style={style.textSeparetor}>Creation for you</Text>
        <CreationList />
      </ScrollView>
    </SafeAreaView>
    );
}

const style = StyleSheet.create({
  contanier:{
    backgroundColor: "#fff", 
    flex: 1 
  },
  textSeparetor:{
    fontSize:20, 
    fontWeight: 'bold', 
    marginLeft:20, 
    marginTop:20
  },
});
