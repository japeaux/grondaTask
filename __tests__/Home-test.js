import React from "react";  
import { create } from "react-test-renderer";
import { Home } from "../src/screens/Home";

const tree = create(<Home/>);
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

test('snapshot', ()=>{
    expect(tree).toMatchSnapshot();
})