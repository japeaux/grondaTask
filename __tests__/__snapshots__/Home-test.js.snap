// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/gronda_web_logo_medium_caribbean.png",
        }
      }
      style={
        Object {
          "height": 80.03999999999999,
          "resizeMode": "contain",
          "width": 375,
        }
      }
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/marco_mueller_banner.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 266.8,
              "marginTop": 10,
              "resizeMode": "contain",
              "width": 675,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 20,
            "marginTop": 20,
          }
        }
      >
        Creation for you
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/JwOEa4hmYk/stories/vf.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  The Limit is The Sky 
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/JwOEa4hmYk/stories/vf.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  Avocado Cream
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/JwOEa4hmYk/stories/vf.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  Sandwich
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/JwOEa4hmYk/stories/q2b.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/W6eu9RP1pc/stories/thumbnail_ik.png",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  Panificando
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/An4jAVSUOQ/stories/qz.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "50%",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://d2bn8tb344j6vy.cloudfront.net/4BHK6UC1U1/stories/8l.JPEG",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "contain",
                      "width": 300,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "padding": 0,
                      "textAlign": "left",
                    }
                  }
                >
                  Desayuno
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
